ml/src/main/scala/org/apache/spark/ml/h2o/models/H2OMOJOModel.scala

```diff
@@ -161,7 +161,7 @@ class H2OMOJOModel(override val uid: String) extends H2OMOJOModelBase[H2OMOJOMod
     val flattenedDF = H2OSchemaUtils.flattenDataFrame(dataset.toDF())
     val relevantColumnNames = flattenedDF.columns.intersect(getFeaturesCols())
     val args = relevantColumnNames.map(flattenedDF(_))
-    flattenedDF.select(col("*"), getModelUdf()(struct(args: _*)).as(getPredictionCol()))
```

Comment:
Good spot. Yes, it does (since Spark 2.0.0). see the [link to 2.1 release](https://github.com/apache/spark/blob/v2.1.0/sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala#L1852)

Reply:
In that case all good, thanks!